clowder-framework/clowder2
# Enable/disable listeners

Addressed:
1. Adding a page for listing all the extractors
2. Only admins should be able to enable/disable them.
3. Regular users should see the enabled extractors only.

To test:
1. run the test_extractors
2. Try to switch to admin_mode and then enable/disable the extractors
3. Then try to switch back to regular mode and you should see only enabled extractors.
4. Try to disable an extractor, then try to submit a job being a regular user from backend(Postman), you should see Forbidden error.

Adding some screenshots for reference.
<img width="1719" alt="1" src="https://github.com/clowder-framework/clowder2/assets/22405928/6f045fda-0e86-4553-ac2e-68297a29efb7">
<img width="1419" alt="2" src="https://github.com/clowder-framework/clowder2/assets/22405928/12ad8de3-366a-49ad-98b9-4b54970bf25f">
<img width="1677" alt="3" src="https://github.com/clowder-framework/clowder2/assets/22405928/9daa672c-6b9a-44f1-9df2-a47034edef6b">
<img width="1681" alt="4" src="https://github.com/clowder-framework/clowder2/assets/22405928/ff8d4fb3-5857-4736-a306-c2c586694f51">
<img width="1392" alt="Screenshot 2024-05-02 at 5 16 16 PM" src="https://github.com/clowder-framework/clowder2/assets/22405928/6292a512-8980-47c5-9f97-ee6f26222848">
<img width="875" alt="Screenshot 2024-05-06 at 3 38 38 PM" src="https://github.com/clowder-framework/clowder2/assets/22405928/661e6d4a-200a-4aa9-b8ee-7879be26c8eb">
